src/node_worker.cc

```diff
@@ -775,6 +784,13 @@ void Worker::Exit(ExitCode code,
                   const char* error_code,
                   const char* error_message) {
   Mutex::ScopedLock lock(mutex_);
+
+  // Remove reference to the port used for inter-thread communication
+  {
+    Mutex::ScopedLock message_ports_lock(messagePortsMutex);
```

Comment:
consider to move this scope before the `Mutex::ScopedLock lock(mutex_);` to avoid unneeded nesting of locks.